core/pkg/sync/file/fileinfo_watcher.go

```diff
@@ -99,27 +103,26 @@ func (f *fileInfoWatcher) Errors() chan error {
 	return f.erChan
 }
 
-// Run is a blocking function that starts the filewatcher's timer thread
-func (f *fileInfoWatcher) Run(ctx context.Context, s time.Duration) error {
+// run is a blocking function that starts the filewatcher's timer thread
+func (f *fileInfoWatcher) run(ctx context.Context, s time.Duration) {
 	// timer thread
-	f.eg.Go(func() error {
+	go func() {
```

Comment:
@Kavindu-Dodan  I think this commit sorts out the goroutine lifecycle and error returns for the fileinfo watcher thread.  I have:

- nixed errgroup
- wired all returned errors during the runtime into the watcher's errchan
- unexported run and moved it into the watcher's constructor.  

Because the watcher is instantiated in source.Init() it is passed a context and uses that as a signal to cleanly exit.  The deferred Close() in the sync code takes care of closing its channels.

Reply:
Yeah, the rework looks good 👍